pygmt/src/grdhisteq.py

```diff
@@ -0,0 +1,367 @@
+"""
+grdhisteq - Perform histogram equalization for a grid.
+"""
+import warnings
+
+import numpy as np
+import pandas as pd
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import (
+    GMTTempFile,
+    build_arg_string,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+from pygmt.io import load_dataarray
+
+
+class grdhisteq:  # pylint: disable=invalid-name
+    r"""
+    Perform histogram equalization for a grid.
+
+    Two common use cases of :meth:`pygmt.grdhisteq` are to find data values
+    that divide a grid into patches of equal area
+    (:meth:`pygmt.grdhisteq.compute_bins`) or to write a grid with
+    statistics based on some kind of cumulative distribution function using
+    (:meth:`pygmt.grdhisteq.equalize_grid`).
+
+    Histogram equalization provides a way to highlight data that has most
+    values clustered in a small portion of the dynamic range, such as a
+    grid of flat topography with a mountain in the middle. Ordinary gray
+    shading of this grid (using :meth:`pygmt.Figure.grdimage` or
+    :meth:`pygmt.Figure.grdview`) with a linear mapping from topography to
+    graytone will result in most of the image being very dark gray, with the
+    mountain being almost white. :meth:`pygmt.grdhisteq.compute_bins` can
+    provide a list of data values that divide the data range into divisions
+    which have an equal area in the image [Default is 16 if ``divisions`` is
+    not set]. The :class:`pandas.DataFrame` or ASCII file output can be used to
+    make a colormap with :meth:`pygmt.makecpt` and an image with
+    :meth:`pygmt.Figure.grdimage` that has all levels of gray occuring
+    equally.
+
+    :meth:`pygmt.grdhisteq.equalize_grid` provides a way to write a grid with
+    statistics based on a cumulative distribution function. In this
+    application, the ``outgrid`` has relative highs and lows in the same
+    (x,y) locations as the ``grid``, but the values are changed to reflect
+    their place in the cumulative distribution.
+    """
+
+    @staticmethod
+    @fmt_docstring
+    @use_alias(
+        C="divisions",
+        D="outfile",
+        G="outgrid",
+        R="region",
+        N="gaussian",
+        Q="quadratic",
+        V="verbose",
```

Comment:
```suggestion
        V="verbose",
        h="header",
```

Reply:
``header`` won't be accessible from ``compute_bins`` yet but could be added in a separate PR when ``output_type="file"``.